docs/overview.rst

```diff
@@ -1,11 +1,410 @@
+
+Restricting sources
+^^^^^^^^^^^^^^^^^^^
+
+In some cases (e.g. unsupervised learning), you might want to use a subset of
+the provided sources. This is achieved by passing a ``sources`` argument to the
+dataset constructor. Here's an example:
+
+>>> restricted_dataset = IndexableDataset(
+...     indexables=OrderedDict([('features', features), ('targets', targets)]),
+...     axis_labels=OrderedDict([('features', ('batch', 'height', 'width')),
+...                              ('targets', ('batch', 'index'))]),
+...     sources=('features',))
+>>> state = restricted_dataset.open()
+>>> print(restricted_dataset.get_data(state=state, request=[0, 1]))
+(array([[[ 47, 211],
+        [ 38,  53]],
+<BLANKLINE>
+       [[204, 116],
+        [152, 249]]]),)
+>>> restricted_dataset.close(state=state)
+
+You can see that in this case only the features are returned by
+:meth:`get_data`.
+
+Iteration schemes: which examples to visit
+------------------------------------------
+
+.. admonition:: In summary
+    :class: tip
+
+    * :class:`IterationScheme` subclasses are responsible for deciding in which
+      order examples are visited.
+    * :meth:`get_request_iterator` returns an iterator object that returns
+      requests.
+    * These requests can be fed to a dataset's :meth:`get_data` method.
+    * Common batchwise iteration schemes are :class:`SequentialScheme` and
+      :class:`ShuffledScheme`.
+    * Common examplewise iteration schemes are :class:`SequentialExampleScheme`
+      and :class:`ShuffledExampleScheme`.
+
+Encapsulating and accessing our data is good, but if we're to integrate it into
+a training loop, we need to be able to iterate over the data. For that, we need
+to decide *which* indices to request and in *which order*. This is accomplished
+via an :class:`IterationScheme` subclass.
+
+At its most basic level, an iteration scheme is responsible, through its
+:meth:`get_request_iterator` method, for building an iterator that will return
+requests. Here are some examples:
+
+>>> from fuel.schemes import (SequentialScheme, ShuffledScheme,
+...                           SequentialExampleScheme, ShuffledExampleScheme)
+>>> schemes = [SequentialScheme(examples=8, batch_size=4),
+...            ShuffledScheme(examples=8, batch_size=4),
+...            SequentialExampleScheme(examples=8),
+...            ShuffledExampleScheme(examples=8)]
+>>> for scheme in schemes:
+...     print(list(scheme.get_request_iterator()))
+[[0, 1, 2, 3], [4, 5, 6, 7]]
+[[7, 2, 1, 6], [0, 4, 3, 5]]
+[0, 1, 2, 3, 4, 5, 6, 7]
+[7, 2, 1, 6, 0, 4, 3, 5]
+
+We can therefore use an iteration scheme to visit a dataset in some order.
+
+>>> state = dataset.open()
+>>> scheme = ShuffledScheme(examples=dataset.num_examples, batch_size=4)
+>>> for request in scheme.get_request_iterator():
+...     data = dataset.get_data(state=state, request=request)
+...     print(data[0].shape, data[1].shape)
+(4, 2, 2) (4, 1)
+(4, 2, 2) (4, 1)
+>>> dataset.close(state)
+
```

Comment:
I guess not all the `iterationScheme`s work with every `Dataset`, e.g. I would expect `ShuffledScheme` not to work with a sequential `IterableDataset`. You could mention it here (no need to add it also to the summary IMHO).